state/factory/factory.go

```diff
@@ -173,8 +143,19 @@ func NewFactory(cfg config.Config, opts ...Option) (Factory, error) {
 func (sf *factory) Start(ctx context.Context) error {
 	sf.mutex.Lock()
 	defer sf.mutex.Unlock()
+	var err error
+	if err = sf.lifecycle.OnStart(ctx); err != nil {
```

Comment:
Do you need to remove L126-L128? Otherwise, sf.accountTrie will be started as well by this call?

Reply:
addressed